src/column_type.rs

```diff
@@ -0,0 +1,89 @@
+use pyo3::prelude::{PyModule, PyModuleMethods};
+use pyo3::{Bound, Py, PyAny, PyResult, Python, pyclass, pymodule};
+
+
+#[pyclass(subclass)]
+pub struct PyNativeType;
+
+#[pyclass(extends=PyNativeType)]
+pub struct Double;
+
+#[pyclass(extends=PyNativeType)]
+pub struct Text;
+
+#[pyclass(extends=PyNativeType)]
+pub struct Int;
+
+#[pyclass(extends=PyNativeType)]
+pub struct Float;
+
+#[pyclass(extends=PyNativeType)]
+pub struct Boolean;
+
+#[pyclass(extends=PyNativeType)]
+pub struct BigInt;
+
+#[pyclass(subclass)]
+pub struct PyCollectionType {
+    #[pyo3(get)]
+    pub frozen: bool,
+}
+
+#[pyclass(extends=PyCollectionType)]
+pub struct Map {
+    #[pyo3(get)]
+    pub key_type: Py<PyAny>,
+    #[pyo3(get)]
+    pub value_type: Py<PyAny>,
+}
```

Comment:
Please try to define a common base class like ColumnType that could be used here instead of PyAny.